### PR TITLE
Fix npmignore

### LINKED ISSUE #1247: Installing js-beautify fails
# Description

I am trying to install a library, which is dependant on this one, and starting today (I've tried the last week the last time) I get the following error when installing my dependencies:

```
npm ERR! path /Users/daniel.rotter/Development/massiveart/sulu-minimal/vendor/sulu/sulu/node_modules/js-beautify/js/bin/css-beautify.js
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall chmod
npm ERR! enoent ENOENT: no such file or directory, chmod '/Users/daniel.rotter/Development/massiveart/sulu-minimal/vendor/sulu/sulu/node_modules/js-beautify/js/bin/css-beautify.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/daniel.rotter/.npm/_logs/2017-09-18T05_55_47_583Z-debug.log
```

It looks to me like this is somehow connected to the latest 1.7.0 release... Any ideas?

# Steps to Reproduce

Create a new folder and install this library as dependency using `npm install js-beautify`.

## Environment
OS: Mac OSX 10.11.6

Deleted most of the PR template, since it's not relevant.

### PR DESCRIPTION
**Pull request by maael**
This fixes the npmignore to also add a rule to exclude the children in js/ from being ignored, so that they will be added when npm packs the package.

`.npmignore` is based on the same rules as `.gitignore`, meaning `!js/` will only exclude the immediate children in `js/`, not any others. This just adds a glob of `!js/**/*` to include the children, and the files needed for installing this package.

Fixes #1247.

### CODE REVIEW

**Comment by simondel** [comment]
My bad sorry! Thanks for fixing it

**Comment by maael** [comment]
No problem, these things happen. 👍

**Comment by ob1** [comment]
@simondel @bitwiseman To fix the internet (by merging this PR and republishing), this needs an approval review from someone with write access

**Comment by simondel** [comment]
@ob1 @bitwiseman So, any idea who would have permissions to merge PR's?

**Comment by maael** [comment]
@simondel from what I can see it looks like only @bitwiseman can merge (although I could be mistaken). If I'm right, this probably won't get looked at for a while at it's 5am in Seattle.

**Comment by andreineculau** [comment]
isn't this a pure liability - that in a project with 5k stars, which so many depend on, only one person has write access? I can literally feel how everyone is keeping their fingers crossed that @bitwiseman doesn't have a hangover :), or his broadband cut off, or, or, or.

https://felixge.de/2013/03/11/the-pull-request-hack.html

**Comment by irvingswiftj** [comment]
@andreineculau , you are right, its like the leftpad story, just by accident this time!

**Comment by szeck87** [comment]
Hmm, wonder how the Travis and Appveyor builds still completed without any errors.  Is it a specific version of node and/or npm?

**Comment by evocateur** [comment]
@bitwiseman: I locally bumped the package.json version to 1.7.1 and published it (with these changes) to unblock folks, so you'll need to update that in the source, too.

File: .npmignore [hunk 1]
```diff
@@ -1,4 +1,8 @@
-**/*
+# Ignore all files
+*
+
+# Add exceptions
 !js/
+!js/**/*
 !CONTRIBUTING.MD
```
Comment (evocateur) [inline .npmignore]: Not sure if we _really_ need `js/test/**` included in the published artifact, but it only adds ~50kb so I didn't bother re-ignoring it.

**Comment by bitwiseman** [comment]
@evocateur You beat me to publishing by about 1 minute. I've published 1.7.2 that also reverts.

**Comment by bitwiseman** [comment]
@maael - Thanks for providing the fix.  I've updated it with an additional admonition.  😄 
@evocateur - Thanks for taking action to mitigate problem.  I've added you as a contributor to the project

**Comment by shawninder** [comment]
@bitwiseman You da man!

**Pull request merged**